services/anvil/anvil.go

```diff
@@ -0,0 +1,102 @@
+package anvil
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os/exec"
+
+	"github.com/ethereum-optimism/mocktimism/service-discovery"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type AnvilService struct {
+	hostname    string
+	port        int
+	serviceType string
+	id          string
+	config      map[string]string
+	cmd         *exec.Cmd
+	logger      log.Logger
+}
+
+func NewAnvilService(logger log.Logger) *AnvilService {
+	return &AnvilService{
+		hostname:    "127.0.0.1",
+		port:        8545,
+		serviceType: "_anvil._tcp",
+		id:          "anvilL1",
+		config:      map[string]string{},
+		logger:      logger,
+	}
+}
+
+func (a *AnvilService) Hostname() string {
+	return a.hostname
+}
+
+func (a *AnvilService) Port() int {
+	return a.port
+}
+
+func (a *AnvilService) ServiceType() string {
+	return a.serviceType
+}
+
+func (a *AnvilService) ID() string {
+	return a.id
+}
+
+func (a *AnvilService) Config() servicediscovery.ServiceConfig {
+	return a.config
+}
```

Comment:
I'm curious if these are needed, or if accessing the properties on `AnvilService` is enough

Reply:
Currently not needed in this specific case but future services will be returning stuff like this lazily from whatever it's proxying too. Since we want the interface to never change we need the more flexible api